src/main/java/bio/terra/service/snapshot/flight/create/SnapshotCreateFlight.java

```diff
@@ -366,6 +367,12 @@ public SnapshotCreateFlight(FlightMap inputParameters, Object applicationContext
 
     addStep(new CreateSnapshotPolicyStep(policyService, sourceDataset.isSecureMonitoringEnabled()));
 
+    // starts a flight to add data access control groups to the snapshot
+    List<String> dataAccessControlGroups = snapshotReq.getDataAccessControlGroups();
```

Comment:
I wonder if there's a validation we can do on the groups list, to avoid creating the snapshot only to have it fail when trying to set the DAC in sam.

Reply:
I'll look into this! Like we could call sam to make sure the groups exist already before trying to add it to the snapshot.

Reply:
Since we want to do the fastest and easiest thing now, I will add a ticket to the backlog to do this work later.